enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -566,6 +629,110 @@ func (p *PebbleCache) updateAtime(key filestore.PebbleKey) error {
 	return db.Set(keyBytes, protoBytes, &pebble.WriteOptions{Sync: false})
 }
 
+func (p *PebbleCache) migrateData(quitChan chan struct{}) error {
+	if len(p.migrators) == 0 {
+		log.Debugf("No migrations necessary")
+		return nil
+	}
+
+	limiter := rate.NewLimiter(rate.Limit(*migrationQPSLimit), 1)
+	db, err := p.leaser.DB()
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	iter := db.NewIter(&pebble.IterOptions{
+		LowerBound: keys.MinByte,
+		UpperBound: keys.MaxByte,
+	})
+	defer iter.Close()
+
+	minVersion := p.maxDatabaseVersion()
+	maxVersion := p.minDatabaseVersion()
+	migrationStart := time.Now()
+	keysSeen := 0
+	keysMigrated := 0
+	lastStatusUpdate := time.Now()
+
+	for iter.First(); iter.Valid(); iter.Next() {
+		if bytes.HasPrefix(iter.Key(), SystemKeyPrefix) {
+			continue
+		}
+		keysSeen += 1
+
+		select {
+		case <-quitChan:
+			return nil
+		default:
+		}
+		_ = limiter.Wait(p.env.GetServerContext())
+
+		if time.Since(lastStatusUpdate) > 10*time.Second {
+			log.Infof("Pebble Cache: data migration progress: saw %d keys, migrated %d to version: %d in %s", keysSeen, keysMigrated, maxVersion, time.Since(migrationStart))
```

Comment:
Can you add iter.Key() to the log message? Since the keys are sorted in Pebble, it gives you some idea of how far along the scan is.